File: android/app/src/main/java/com/github/makewheels/video2022/data/model/Comment.kt
package com.github.makewheels.video2022.data.model

data class Comment(
    val id: String,
    val videoId: String?,
    val userId: String?,
    val userPhone: String?,
    val content: String?,
    val parentId: String?,
    val replyToUserId: String?,
    val replyToUserPhone: String?,
    val likeCount: Int?,
    val replyCount: Int?,
    val createTime: String?,
    val updateTime: String?
)

data class CommentPageResponse(
    val list: List<Comment>,
    val total: Long,
    val totalPages: Int,
    val currentPage: Int,
    val pageSize: Int
)

data class AddCommentRequest(
    val videoId: String,
    val content: String,
    val parentId: String? = null
)

data class CommentCount(
    val count: Int
)

File: ios/VideoApp/Sources/Models/Comment.swift
import Foundation

struct Comment: Decodable, Identifiable {
    let id: String
    let videoId: String?
    let userId: String?
    let userPhone: String?
    let content: String?
    let parentId: String?
    let replyToUserId: String?
    let replyToUserPhone: String?
    let likeCount: Int?
    let replyCount: Int?
    let createTime: String?
    let updateTime: String?
}

struct CommentPage: Decodable {
    let list: [Comment]
    let total: Int
    let totalPages: Int
    let currentPage: Int
    let pageSize: Int
}

struct AddCommentRequest: Encodable {
    let videoId: String
    let content: String
    let parentId: String?
}

struct CommentCount: Decodable {
    let count: Int
}
